tests/common.go

```diff
@@ -10740,3 +10740,157 @@ func CreateNFSProxyStorageClass(scName, nfsServer, mountPath string) error {
+						if status.Status != v1.ConditionTrue && stNodeNames[k8sNode.Name] {
+							nodeNotReadyeCount += 1
+						}
+						break
+					}
+				}
+
+			}
+			if nodeNotReadyeCount > 1 {
+				err = fmt.Errorf("multiple  nodes are Unschedulable at same time,"+
+					"node status:%#v", nodeSchedulableStatus)
+				log.Errorf("Got error : %s", err.Error())
+				log.Infof("Node Details: %#v", nodeList.Items)
+				output, err := Inst().N.RunCommand(stNodes[0], "pxctl status", node.ConnectionOpts{
+					IgnoreError:     false,
+					TimeBeforeRetry: defaultRetryInterval,
+					Timeout:         defaultTimeout,
+					Sudo:            true,
+				})
+				if err != nil {
+					log.Errorf("failed to get pxctl status, Err: %v", err)
+				}
+				log.Infof(output)
+				*mError = err
+				return
+			}
+		}
+		itr++
+		time.Sleep(30 * time.Second)
+	}
+}
+
+// printK8sClusterInfo prints info about K8s cluster nodes
+func PrintK8sCluterInfo() {
+	log.Info("Get cluster info..")
+	t := func() (interface{}, bool, error) {
+		nodeList, err := core.Instance().GetNodes()
+		if err != nil {
+			return "", true, fmt.Errorf("failed to get nodes, Err %v", err)
+		}
+		if len(nodeList.Items) > 0 {
+			for _, node := range nodeList.Items {
+				nodeType := "Worker"
+				if core.Instance().IsNodeMaster(node) {
+					nodeType = "Master"
+				}
+				log.Infof(
+					"Node Name: %s, Node Type: %s, Kernel Version: %s, Kubernetes Version: %s, OS: %s, Container Runtime: %s",
+					node.Name, nodeType,
+					node.Status.NodeInfo.KernelVersion, node.Status.NodeInfo.KubeletVersion, node.Status.NodeInfo.OSImage,
+					node.Status.NodeInfo.ContainerRuntimeVersion)
+			}
+			return "", false, nil
+		}
+		return "", false, fmt.Errorf("no nodes were found in the cluster")
+	}
+	if _, err := task.DoRetryWithTimeout(t, 1*time.Minute, 5*time.Second); err != nil {
+		log.Warnf("failed to get k8s cluster info, Err: %v", err)
+	}
+}
+
+//ExportSourceKubeConfig changes the KUBECONFIG environment variable to the source cluster config path
+func ExportSourceKubeConfig() error {
+	sourceClusterConfigPath, err := GetSourceClusterConfigPath()
+	if err != nil {
+		return err
+	}
+	err = os.Unsetenv("KUBECONFIG")
+	if err != nil {
+		return err
+	}
+	return os.Setenv("KUBECONFIG", sourceClusterConfigPath)
+}
+
+//ExportDestinationKubeConfig changes the KUBECONFIG environment variable to the destination cluster config path
+func ExportDestinationKubeConfig() error {
+	DestinationClusterConfigPath, err := GetDestinationClusterConfigPath()
+	if err != nil {
+		return err
+	}
+
+	err = os.Unsetenv("KUBECONFIG")
+	if err != nil {
+		return err
+	}
+	return os.Setenv("KUBECONFIG", DestinationClusterConfigPath)
+}
+
+//SwitchBothKubeConfigANDContext switches both KUBECONFIG and context to the given cluster
+func SwitchBothKubeConfigANDContext(cluster string) error {
+	if cluster == "source" {
+		err := ExportSourceKubeConfig()
+		if err != nil {
+			return err
+		}
+		err = SetSourceKubeConfig()
```

Comment:
How about including ExportSourceKubeConfig() within SetSourceKubeConfig()?

Reply:
It's an expensive op why do it when it's not required? This is only required as currently the schedular upgrade actually runs oc commands instead of using the api way. Once we fix that we don't need it.